### PR TITLE
feat(objectstorage): add pure object storage without the operating of object storage metadata and predefined objectName

### PR DESCRIPTION
**Pull request by CHLK**
What type of PR is this?
type-feature

What this PR does / why we need it:
add pure object storage without the operating of object storage metadata and predefined objectName

### CODE REVIEW

**Comment by CHLK** [comment]
@yizhouxw I've implemented your suggestion, please take a look when you have a chance.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/objectstorage/client/CloudObjectStorageClient.java [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+
+    public CloudObjectStorageClient(CloudObjectStorage publicEndpointCloudObjectStorage,
+            CloudObjectStorage internalEndpointCloudObjectStorage,
+            ObjectStorageConfiguration objectStorageConfiguration) {
+        this.publicEndpointCloudObjectStorage = publicEndpointCloudObjectStorage;
+        this.internalEndpointCloudObjectStorage = internalEndpointCloudObjectStorage;
+        this.objectStorageConfiguration = objectStorageConfiguration;
+        if (this.publicEndpointCloudObjectStorage.supported()) {
+            validateBucket();
+            log.info("Cloud pure object storage initialized");
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/objectstorage/client/CloudObjectStorageClient.java:85]: log content does not match class name

File: server/odc-service/src/main/java/com/oceanbase/odc/service/objectstorage/client/ObjectStorageClient.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ * only for the logic of object storage ,not include the processing of metadata.
+ * </p>
+ * In addition, the bucketName uses the default value and cannot be customized
+ * 
+ * @author keyang
+ * @date 2024/08/09
+ * @since 4.3.2
+ */
+public interface ObjectStorageClient {
+    URL getDownloadUrl(@NotBlank String objectName, Long expirationSeconds);
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/objectstorage/client/ObjectStorageClient.java:39]: generateDownloadUrl, 
generateUploadUrl

File: server/odc-service/src/main/java/com/oceanbase/odc/service/objectstorage/cloud/CloudObjectStorage.java [hunk 3]
```diff
@@ -59,6 +59,8 @@ public interface CloudObjectStorage {
 
     PutObjectResult putObject(String bucketName, String key, File file, ObjectMetadata metadata) throws CloudException;
 
+    void setExpiredAfterLastModified(String bucketName, String prefixForMatchObjectName, int expiredDaya);
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/objectstorage/cloud/CloudObjectStorage.java:62]: unused code, may remove it

**Review by yizhouxw — APPROVED** [review]
lgtm

**Pull request merged**